srt-protocol/src/packet/control.rs

```diff
@@ -1526,4 +1522,68 @@ mod test {
             )
         );
     }
+
+    #[test]
+    fn test_unordered_hs_extensions() {
+        //Taken from Wireshark dump of FFMPEG connection handshake
```

Comment:
Would be great if you included the full commandline

Reply:
This connection was initiated using OBS Studio, which uses ffmpeg via API calls.
The connection string used in OBS was
```
srt://localhost:9710?latency=60000&mss=1464&streamid=#!::u=hex&passphrase=9LS6wkXLpBo&pbkeylen=32&transtype=live
```
(same as in the PR description).
Should I just add that as a comment, or do you need me to recreate this on ffmpeg's cli?

Reply:
(BTW: No worries about the passphrase, it's just a testing value for local connections)

Reply:
OBS does start a subprocess for this connection, but using its own ffmpeg cli:
```
obs-ffmpeg-mux.exe" "srt://localhost:9710?latency=60000&mss=1464&streamid=#!::u=hex&passphrase=9LS6wkXLpBo&pbkeylen=32&transtype=live" 1 1 h264 2000 1920 1080 1 1 1 1 30 1 aac "adv_stream_aac" 160 48000 2 "" "" 
```

Reply:
If it's not simple, it's fine how it is. I'll go ahead and merge.